hack/ci-e2e.sh

```diff
@@ -32,58 +43,100 @@ minikube start
+
+virt-install \
+   --connect qemu:///system \
+   --name "${VM_NAME}" \
+   --description "Virtualized BareMetalHost" \
+   --osinfo=ubuntu-lts-latest \
+   --ram=4096 \
+   --vcpus=2 \
+   --disk size=20 \
+   --graphics=none \
+   --console pty \
+   --serial pty \
+   --pxe \
+   --network network=baremetal-e2e,mac="${BOOT_MAC_ADDRESS}" \
+   --noautoconsole
 
-# Add BMH VM to VBMC
-docker exec vbmc vbmc add "${VM_NAME}" --port "${VBMC_PORT}"
-docker exec vbmc vbmc start "${VM_NAME}"
-docker exec vbmc vbmc list
-
-# These variables are used by the tests. They override variables in the config file.
 # This IP is defined by the network we created above.
-# Together with the VBMC_PORT this becomes the BMC_ADDRESS used by the BMH in the test.
 IP_ADDRESS="192.168.222.1"
-export BMC_ADDRESS="ipmi://${IP_ADDRESS}:${VBMC_PORT}"
-export BOOT_MAC_ADDRESS
+
+# These variables are used by the tests. They override variables in the config file.
 # These are the VBMC defaults (used since we did not specify anything else for `vbmc add`).
 export BMC_USER=admin
 export BMC_PASSWORD=password
-CIRROS_VERSION="0.6.2"
-IMAGE_FILE="cirros-${CIRROS_VERSION}-x86_64-disk.img"
-export IMAGE_CHECKSUM="c8fc807773e5354afe61636071771906"
-export IMAGE_URL="http://${IP_ADDRESS}/${IMAGE_FILE}"
-IMAGE_FOLDER="${REPO_ROOT}/test/e2e/images"
 
-## Setup image server
-# Create a directory for images
+if [[ "$USE_TOOL" == "vbmc" ]]; then
+	# Start VBMC
+	docker run --name vbmc --network host -d \
+	-v /var/run/libvirt/libvirt-sock:/var/run/libvirt/libvirt-sock \
+	-v /var/run/libvirt/libvirt-sock-ro:/var/run/libvirt/libvirt-sock-ro \
+	quay.io/metal3-io/vbmc
+
+	VBMC_PORT="16230"
+	# Together with the VBMC_PORT this becomes the BMC_ADDRESS used by the BMH in the test.
+	export BMC_ADDRESS="ipmi://${IP_ADDRESS}:${VBMC_PORT}"
+
+	# Add BMH VM to VBMC
+	docker exec vbmc vbmc add "${VM_NAME}" --port "${VBMC_PORT}"
+	docker exec vbmc vbmc start "${VM_NAME}"
+	docker exec vbmc vbmc list
+
+
+elif [[ "$USE_TOOL" == "sushy-tools" ]]; then
+	# Sushy-tools variables
+	SUSHY_EMULATOR_FILE="${REPO_ROOT}"/test/e2e/sushy-tools/sushy-emulator.conf
+	SUSHY_PORT="8000"
+
+	# Start sushy-tools
+	# Privileged mode is needed to access libvirt.
+	sudo docker run --name sushy-tools -d --privileged --network host \
+	-v "${SUSHY_EMULATOR_FILE}":/etc/sushy/sushy-emulator.conf:Z \
+	-v /var/run/libvirt:/var/run/libvirt:Z \
+	-e SUSHY_EMULATOR_CONFIG=/etc/sushy/sushy-emulator.conf \
+	quay.io/metal3-io/sushy-tools:latest sushy-emulator
+
+	sleep 5
+
+	# Fetch the ID of the system with the name 'bmo-e2e-0'
+	ids=$(curl -s "http://${IP_ADDRESS}:${SUSHY_PORT}/redfish/v1/Systems" | jq -r '.Members[] | .["@odata.id"] | split("/")[-1] | rtrimstr("\"")')
+
+	# Iterate over each ID and fetch its details
+	for id in $ids; do
+		data=$(curl -s "http://${IP_ADDRESS}:${SUSHY_PORT}/redfish/v1/Systems/${id}")
+		name=$(echo "$data" | jq -r '.Name')
+		if [[ "$name" == "bmo-e2e-0" ]]; then
+			DESIRED_ID="$id"
```

Comment:
```suggestion
			DESIRED_ID="${id}"
```
Also, why are we saving this id into DESIRED_ID, if we're only using it on the next line, and then breaking?

Reply:
Yes I see what you mean and you're right, I'll change it, it's a relic from an old fix.